### ISSUE TITLE
[Feature request] Set image size per block

### ISSUE THREAD

**Issue by WilliamCappelletti** [thequilo/obsidian-tikz-renderer#4]
## Motivation

Some images are too big compared to text and per-image setting would be very useful.

## Desired behavior

Obtain the same result as `<img src="image.png" style="width: 50%;" />` by writing something like 
```
    ```latex tikz render 50%
    \draw (0,0) -- (1,0) -- (1,1) -- (0,1) -- cycle;
    ```
```
Or with absolute size in pixel `![[image.png|250]]` (obsidian syntax):
```
    ```latex tikz render 250
    \draw (0,0) -- (1,0) -- (1,1) -- (0,1) -- cycle;
    ```
```

**Comment by thequilo** [comment]
I agree! The image size is currently set by a very crude CSS class. But I'm not sure about the syntax for this. I don't think that Markdown supports something like this natively. We could find inspiration from emacs org mode code blocks or allow general styling.

- Add a syntax for keyword arguments: `latex tikz render width:250`, or
- add general styling support: `latex tikz render {width: 50%}`

**Comment by WilliamCappelletti** [comment]
I think the second one would be the most flexible, and the easiest to implement.
You could just parse whatever is in `{<style>}` and append it to your `tikz-render-container tikz-preview tikz-preview-image-only`:
```html
<div class="tikz-render-container tikz-preview tikz-preview-image-only" style="<style>">
  ...
</div>
```

**Comment by thequilo** [comment]
I finally found time to look at this issue. I face a problem for which I don't know a solution: The postprocessor is only invoked when the content of the code block changes. If the "header line" is modified, the preview re-uses a cached version of the rendered HTML. This means that, for the style to have a visible effect, you always have to modify the code block contents as well, which is not very comfortable.

This issue is also present when modifying the "header" line from `latex` to `latex tikz render`.

An alternative could be to allow special comments in the code blocks, e.g., `%obsidian-tikz-style: {width: 100px}` in the first line of the code block.

**Comment by WilliamCappelletti** [comment]
The issue you faced is also a limitation for the `side-by-side` and the `image-only` mode: you cannot toggle between them without changing the content.

**Comment by ghost** [comment]
I think I implemented this (**partially**. It can't enlarge images, ~~and **it behaves weird in side-by-side mode**) in [my fork of this plugin](https://github.com/BeOnDrag/better-obsidian-tikz-renderer). You can do this by multiply the height (attribute in `<svg>`) by the scale and divide the width by the scale. **Only use this in image-only mode!**~~ Okay I've found a better way.
1. Set the width of svg element to the scale percentage (`svgEl.style.width = scale * 100 + "%"`)
2. Use some styles, like `.tikz-preview { display: flex; justify-contents: space-evenly; }` to align the image
3. Done! It works fine in both modes:
![屏幕截图(67)](https://user-images.githubusercontent.com/92842832/166876036-dc3bbb36-68db-4cde-883e-d0efde1d66f3.png)

And...
> The issue you faced is also a limitation for the side-by-side and the image-only mode: you cannot toggle between them without changing the content.

The fact is Obsidian doesn't update a codeblock's first line if its language is not changed. You can switch `latex tikz render` to `tikz render`, or backwards, to make changes.
